### ISSUE TITLE
Cloud2Edge: Ditto helm chart moved to Ditto's repository

### ISSUE THREAD

**Issue by sgloutnikov** [eclipse/packages#486]
I just noticed that the [Ditto helm chart](https://github.com/eclipse/packages/tree/master/charts/ditto) used in cloud2edge [requirements.yaml](https://github.com/eclipse/packages/blob/master/packages/cloud2edge/requirements.yaml) is no longer maintained in this repository but was moved to [Ditto's repository](https://github.com/eclipse-ditto/ditto/tree/master/deployment/helm/ditto) and is also served from a different Helm repository. Does this mean that the cloud2edge package should be updated to use the new and updated Ditto Helm chart, or are there some incompatibilities that could be preventing the switch over?

I'll test on my end and see how far I can get with updating the cloud2edge Ditto version to latest.

**Comment by casafurix** [comment]
Hello, I am actually unable to install the cloud2edge package using the steps mentioned on the website, can you please guide me the right steps now to install it? I want to use Ditto+Hono. Thanks

**Comment by calohmn** [comment]
(Cloud2edge installation issues were resolved with @casafurix in #481. In general, the [Gitter channel](https://gitter.im/eclipse/packages) or Stackoverflow (with [eclipse-iot](https://stackoverflow.com/questions/tagged/eclipse-iot) tag) are good places for asking questions, or, in case of a chart or documentation bug, an issue can be raised here.)

**Comment by calohmn** [comment]
@sgloutnikov yes, the cloud2edge chart should be updated. A PR is welcome here.

**Comment by calohmn** [comment]
@sgloutnikov The update actually looked quite straightforward. I've integrated it in PR #493 .

**Comment by sgloutnikov** [comment]
Really strange, I'm unable to bring up the environment in a fresh k3s or minikube single node cluster, any tips are greatly appreciated. This behaviour seems identical for k3s with LoadBalancer or minikube without. 

Looks like a bunch of pods failing with rediness probe, which chains down to nginx and swagger-ui not starting up. Lastly the gateway pod rediness probe looks like is missing its host, and only port is being configured. Some details bellow:

Helm install:
```
 cloud2edge git:(pr/493) helm install -n $NS --wait --timeout 20m --set hono.useLoadBalancer=true --set ditto.nginx.service.type=LoadBalancer $RELEASE ./
```

Get pods:
```
 ➜  ~ k get pods
NAME                                           READY   STATUS                  RESTARTS        AGE
c2e-ditto-nginx-6859974f89-5s5gv               0/1     Init:0/1                0               10m
c2e-ditto-dittoui-649d4b487d-jz7nt             1/1     Running                 0               10m
c2e-ditto-gateway-9fc695954-tmb75              0/1     Running                 0               10m
c2e-ditto-thingssearch-76c969667d-mpmqk        0/1     Running                 0               10m
c2e-ditto-policies-56fc485445-pxmqb            0/1     Running                 0               10m
c2e-ditto-things-6944d6d6fd-rzjn7              0/1     Running                 0               10m
c2e-ditto-connectivity-7d94b56b85-85rzt        0/1     Running                 0               10m
c2e-zookeeper-0                                1/1     Running                 0               10m
ditto-mongodb-778c99d5bf-2lsq5                 1/1     Running                 0               10m
c2e-kafka-0                                    1/1     Running                 0               10m
c2e-service-auth-7d7d8dfdf6-p4d84              1/1     Running                 0               10m
c2e-service-device-registry-86484b7f7b-wkjxg   1/1     Running                 0               10m
c2e-service-command-router-587d5c68fd-kg5fl    1/1     Running                 0               10m
c2e-adapter-mqtt-687c66bbdd-548pc              1/1     Running                 0               10m
c2e-adapter-amqp-77579d66f4-hgzt8              1/1     Running                 0               10m
c2e-adapter-http-d56575cbf-kmlkl               1/1     Running                 0               10m
c2e-ditto-swaggerui-646678bdfd-5lkr8           0/1     Init:CrashLoopBackOff   6 (3m36s ago)   10m
```

Get services:
```
 ➜  ~ k get services
NAME                              TYPE           CLUSTER-IP      EXTERNAL-IP    PORT(S)                      AGE
c2e-kafka-headless                ClusterIP      None            <none>         9092/TCP,9093/TCP            10m
c2e-zookeeper-headless            ClusterIP      None            <none>         2181/TCP,2888/TCP,3888/TCP   10m
ditto-mongodb                     ClusterIP      10.43.206.8     <none>         27017/TCP                    10m
c2e-service-device-registry       ClusterIP      10.43.89.217    <none>         5671/TCP,8080/TCP,8443/TCP   10m
c2e-service-auth                  ClusterIP      10.43.125.245   <none>         5671/TCP,8088/TCP            10m
c2e-ditto-swaggerui               ClusterIP      10.43.22.151    <none>         8080/TCP                     10m
c2e-service-command-router        ClusterIP      10.43.133.236   <none>         5671/TCP                     10m
c2e-ditto-dittoui                 ClusterIP      10.43.239.86    <none>         8080/TCP                     10m
c2e-ditto-gateway                 ClusterIP      10.43.142.65    <none>         8080/TCP                     10m
c2e-kafka                         ClusterIP      10.43.32.223    <none>         9092/TCP,9094/TCP            10m
c2e-zookeeper                     ClusterIP      10.43.159.167   <none>         2181/TCP,2888/TCP,3888/TCP   10m
c2e-kafka-0-external              LoadBalancer   10.43.151.26    172.XX.XX.XX   9094:32094/TCP               10m
c2e-adapter-amqp                  LoadBalancer   10.43.163.121   172.XX.XX.XX   5671:31200/TCP               10m
c2e-ditto-nginx                   LoadBalancer   10.43.193.12    172.XX.XX.XX   8080:30341/TCP               10m
c2e-service-device-registry-ext   LoadBalancer   10.43.181.30    172.XX.XX.XX   28443:31396/TCP              10m
c2e-adapter-http                  LoadBalancer   10.43.50.24     172.XX.XX.XX   8443:32470/TCP               10m
c2e-adapter-mqtt                  LoadBalancer   10.43.186.5     172.XX.XX.XX   8883:32334/TCP               10m
```

Describe nginx pod and logs:
```
...
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  12m   default-scheduler  Successfully assigned spg-dev/c2e-ditto-nginx-6859974f89-5s5gv to za23040014-l
  Normal  Pulling    12m   kubelet            Pulling image "rancher/curlimages-curl:7.73.0"
  Normal  Pulled     12m   kubelet            Successfully pulled image "rancher/curlimages-curl:7.73.0" in 3.785419237s (3.785428409s including waiting)
  Normal  Created    12m   kubelet            Created container wait-for-gateway
  Normal  Started    12m   kubelet            Started container wait-for-gateway
  
  
   ~ k logs c2e-ditto-nginx-6859974f89-5s5gv
Defaulted container "ditto-nginx" out of: ditto-nginx, wait-for-gateway (init)
Error from server (BadRequest): container "ditto-nginx" in pod "c2e-ditto-nginx-6859974f89-5s5gv" is waiting to start: PodInitializing
```

Describe gateway pod:
```
Containers:
  ditto-gateway:
    Container ID:   containerd://bcc2029e8d57fa99a7cc8514f6ef97eb9cc743131ad7bb9feedd0db9bd5d57db
    Image:          docker.io/eclipse/ditto-gateway:3.3.6
    ...
    Liveness:   http-get http://:management/alive delay=160s timeout=5s period=10s #success=1 #failure=4
    Readiness:  http-get http://:management/ready delay=30s timeout=3s period=10s #success=1 #failure=3
```

Note missing host and only management port in Livenesa nd Readiness probes.

**Comment by calohmn** [comment]
@sgloutnikov Good point. I actually wasn't testing with `ditto.nginx.service.type=LoadBalancer`. I was testing without any such chart parameters being set (the `hono.useLoadBalancer=true` is the default anyway - we need to adapt the documentation there).

About `ditto.nginx.service.type=LoadBalancer` we need to check whether there's an issue there with the Ditto chart. I actually don't see setting this option to LoadBalancer being documented in the Ditto repo.

**Comment by sgloutnikov** [comment]
@calohmn If I install each chart separately, the Hono chart installed fine. However, the latest Ditto chart has the same problem as above. I am installing with the default command which pulls the latest version: `helm install -n ditto my-ditto oci://registry-1.docker.io/eclipse/ditto --wait`. 

In what environment are you testing the cloud2edge chart that it's able to install properly? I'm unable to get it to install your PR version, with or without extra parameters, on my local single node k3s or minikube.

**Comment by calohmn** [comment]
@sgloutnikov I'm using minikube v1.29.0 (with v1.31.2 I have the issue that pods are stuck in "terminating" state when uninstalling the chart). I'm running `minikube tunnel`.

**Comment by sgloutnikov** [comment]
I'm convinced the installation issues above are specific to my environment, most likely due to DNS inside the Kubernetes cluster and pods being unable to reach each other for the Akka readiness checks. I was able to install the Ditto Hono chart in minikube with a VirtualBox driver when the Docker one before was giving me the above-mentioned problems. Can't pin it down exactly, but will report if I find more.

**Comment by sgloutnikov** [comment]
Resolved in #493
